File: Common/Example03_console_input/c/src/Input04.c
// read all characters from a single line of console input and print them to the console
#include<stdio.h>

int main(){
    char buffer[256];
    printf("Enter a line of text: ");
    fgets(buffer, sizeof(buffer), stdin); // read a line of text
    printf("You entered: %s", buffer);
    return 0;
}

File: Common/Example03_console_input/java/src/Input04.java
// read all characters from a single line of console input and print them to the console
import java.util.Scanner;

public class Input04 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a line of text: ");
        String input = scanner.nextLine();
        System.out.println("You entered: " + input);
        scanner.close();
    }
}